### PR TITLE
The Return of Digbot

### PR DESCRIPTION
**Pull request by MarkNais**
## About the PR 
Revives digbot — a once removed, now restored and polished fab/craftable robot that mines rocks. It can be turned off and on, told to mine or avoid suspicious rocks (object, gem, etc), and told to target a threshold of hardness. 

Animations have been added, there's unique build-step sprites, it has a tiny-blue tooltip UI instead of a chatlog UI, it points to each rock it targets, and it makes a small clang whenever it's deactivated (as it's hover drive shuts off)

Credits:
Original base sprite ???
New animations & overlays - The Gorog - https://github.com/TheGorog

## Why's this needed? 
Helpful to miners, and adds another cute robot to the game.
@UrsulaMejor told me too, and many people were excited for it's return.

## Changelog

```
(u)MarkNstein:
(*)Digbot revived: a bot for miners. Craftable and has a blueprint in robotics and mining fabs.
```
[FEATURE]
bKMBbaBltI
Oijo4TVCXA dreamseeker_TZ9hLr53Pl dreamseeker_3NyTSMSfVR dreamseeker_vluVQBXodc

### CODE REVIEW

**Comment by MarkNais** [comment]
I forgot to reimplement behavior locking in the UI change — But also, not sure if I care to reimplement it, since it's not "dangerous" (it can't trigger explosive-asteroids AFAIK, since it's not a mob). 
I'll likely trim the "access" code next time I'm awake and coding.

**Comment by UrsulaMejor** [comment]
ARISE, DIGBOT

File: code/modules/robotics/bot/mining.dm [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+			else
+				boutput(user,  "You already added that part!")
+				return
+		else if (istype(T, /obj/item/mining_tool/drill))
+			if (src.build_step == 1)
+				if (user.r_hand == T) user.u_equip(T)
+				else user.u_equip(T)
+				boutput(user,  "You add [T.name]. Now you have a finished mining bot! Hooray!")
+				qdel(T)
+				new /obj/machinery/bot/mining/drill(user.loc)
```
Comment (UrsulaMejor) [inline code/modules/robotics/bot/mining.dm:348]: why not get_turf(src)? See my other comment on this

File: code/obj/item/clothing/helmets.dm [hunk 2]
```diff
@@ -384,6 +384,16 @@
 			light_dir.update(0)
 		return
 
+	attackby(var/obj/item/T, mob/user as mob)
+		if(istype(T, /obj/item/device/prox_sensor))
+			boutput(user,  "You attach the proximity sensor to the hard hat. Now you need to add a robot arm.")
+			new /obj/item/digbotassembly(get_turf(user))
```
Comment (UrsulaMejor) [inline code/obj/item/clothing/helmets.dm:390]: why not get_turf(src)? This kinda awkwardly moves it to your tile if you're mass producing digbot assemblies on the floor

**Review by UrsulaMejor — CHANGES_REQUESTED** [review]

File: code/obj/item/clothing/helmets.dm [hunk 3]
```diff
@@ -384,6 +384,16 @@
 			light_dir.update(0)
 		return
 
+	attackby(var/obj/item/T, mob/user as mob)
```
[the remark this answers is not in the document]
Comment (MarkNais) [reply code/obj/item/clothing/helmets.dm:387]: Ultimately I don't care, but I've added a type check to make it hardhats only.

**Comment by MarkNais** [comment]
Recent changes tested, still craftable, and now operates slightly faster.

**Comment by Gannets** [comment]
I think digbot could benefit from a full resprite in the future, but this is great for now!

**Review by UrsulaMejor — APPROVED** [review]

**Comment by UrsulaMejor** [comment]
Mark said he tested it and that's good enough for me!

**RISE, DIGBOT.**

**Pull request merged**